doc/dev/osd_internals/manifest.rst

```diff
@@ -21,43 +20,121 @@ This document exists to detail:
 
 1. Manifest data structures
 2. Rados operations for manipulating manifests.
-3. How those operations interact with other features like snapshots.
+3. Status and Plans
+
+
+Intended Usage Model
+====================
+
+RBD
+---
+
+For RBD, the primary goal is for either an osd-internal agent or a
+cluster-external agent to be able to transparently shift portions
+of the consituent 4MB extents between a dedup pool and a hot base
+pool.
+
+As such, rbd operations (including class operations and snapshots)
+must have the same observable results regardless of the current
+status of the object.
+
+Moreover, tiering/dedup operations must interleave with rbd operations
+without changing the result.
+
+Thus, here is a sketch of how I'd expect a tiering agent to perform
+basic operations:
+
+* Demote cold rbd chunk to slow pool:
+
+  1. Read object, noting current user_version.
+  2. In memory, run CDC implemenation to fingerprint object.
+  3. Write out each resulting extent to an object in the cold pool
+     using the CAS class.
+  4. Submit operation to base pool:
+
+     * ASSERT_VER with the user verion from the read to fail if the
+       object has been mutated since the read.
+     * SET_CHUNK for each of the extents to the corresponding object
+       in the base pool.
+     * EVICT_CHUNK for each extent to free up space in the base pool.
+       Results in each chunk being marked MISSING.
+
+     (QUESTION: EVICT_CHUNK here is new.  I think this would simplify
+     things considerably)
```

Comment:
Hmm, or it could fail if the extent isn't dedup'd -- in such a case presumably it raced with a write and eviction probably isn't desirable.  What do you think?

Reply:
Do you mean the following case ?

Write object A (indicate that this has a extent to be deduped)
EVICT_CHUNK A (return fail because this is not deduped yet, also means that it is dirty)

If so, Is there any problem if we  make use of flush+evict.

Reply:
Yeah, I think that's the plan.  Evict will simply fail if there isn't already a CAS referene.  I'll update the doc.